src/components/map/bike-map.tsx

```diff
@@ -63,14 +64,42 @@ const BikeMap: React.FC<BikeMapProps> = ({ className, isLoggedIn }) => {
       }
 
       bookBike(selectedStation)
-        .then(() => navigate('/bookings'))
+        .then(() => {
+          closePopup();
+          loadStationDetail(selectedStation);
+          setSelectedStation(selectedStation);
+        })
         .catch(err => {
           console.error("Error while reserving bike:", err);
           toast(MAP.POPUP.RENT_DIALOG.ALERT.DEFAULT_ERR, { type: 'error' });
         });
     },
     [selectedStation, MAP],
   );
+  const handleCancelBooking = useCallback(
+    () => {
+      fetchBooking().then(() => {
+        if (!booking) {
+          throw new Error("Trying to cancel a booking, but no bike booked.");
```

Comment:
This should be accepted instead of throwing an error. The goal of canceling the booking is to make it go away, which, when it is already away, is a goal reached, so no real reason for an error here.

Reply:
I thought of it that way: It's an error when the user can request the cancellation of a booking without having one. In the current process, there is no way for an earlier canceling before reaching this code (at least without external events).

I wanted to make the code match the one for `handleBook` and `handleRent` - I can also make `!booking` trigger something like an early return, however I wouldn't be sure about which behavior would be suitable...

Reply:
I'm now sure about how to rewrite it, just tell me if you really want `!booking` an non-error case.

Reply:
External events are _the issue_ though. When the booking is opened close to the 15min limit and then is canceled while the popup is active, the user receives an error that doesn‘t really serve a purpose. :)

Reply:
So yeah, I‘d like !booking to not be an error here.

Reply:
Ok yeah, I had the "popup is still open but the app is offline"-situation in mind, which would led to the `fetchbooking` being resolved a lot later and then causing an error when the app is back online but the booking has already been canceled through another device or by time limit.

Reply:
See 8703487ca6693468e17f0343cc47dc216708cd87.